cgra/util.py

```diff
@@ -141,4 +141,6 @@ def create_core(xx: int, yy: int):
     elif global_signal_wiring == GlobalSignalWiring.ParallelMeso:
         apply_global_parallel_meso_wiring(interconnect, io_sides,
                                           num_cfg=num_parallel_config)
+    if add_pd:
```

Comment:
can we have this as a pass on top? rather than included in this function

Reply:
No. Unless you want to return an unfinalized interconnect object. Another workaround would be moving the finalizing logic to `circuit()` function.